### DISCUSSION TITLE
Retrieve workflow_run_id for run triggered by repository_dispatch event

### DISCUSSION THREAD

**Discussion by mro-pbraswell** [community/community#27226] [Apps, API and Webhooks]
After submitting a `repository_dispatch` event, I would like to get the `workflow_run_id` for the workflow run that was created. Unfortunately the `gh` cli doesn't return this information, nor does the REST API. In the past, I've been able to use the API to list the workflow runs for the workflow and wait until there's a pending run. This wasn't ideal, but it worked.

Now, however, I've got a repository that potentially has several active/pending runs at one time for the same workflow, and as far as I can tell, there's no way to identify the particular run that was just initiated by the `repository_dispatch` event. The event is being sent with a uniquely identifiable `event_type`, but that information doesn't appear to be available from the data that's provided by the REST API or the gh cli.

**Comment by sambshapiro** [comment]
I have the same problem/desire. The closest I've been able to get is changing to a workflow dispatch and following this recommendation... which really sucks. https://stackoverflow.com/questions/69479400/get-run-id-after-triggering-a-github-workflow-dispatch-event

**Reply by mro-pbraswell** [reply]
Wow. That's pretty clever/terrible. Might have to use it though until there's a real solution. 

 Thanks for passing that along.

**Comment by sambshapiro** [comment]
Related: https://github.com/orgs/community/discussions/17389

**Comment by adam-kafene** [comment]
I ran into this scenario today too.   I don't think anything has changed since July 2022 (date of last comment on this post) and Jan 2023.  I'm going to do a little more research tomorrow, before implenting a workaround.

**Comment by adam-kafene** [comment]
Hey folks, 

I don't think GitHub has implemented anything new, however there is a workaround on another GitHub discussion [here](https://github.com/orgs/community/discussions/9752). If you look at the PR listed in there, we can get around this seemingly easier than the workaround listed in this discussion.   That's the route I'm going :shrug:

**Comment by MSevey** [comment]
Found this helpful action as a work around https://github.com/marketplace/actions/return-dispatch
